Dockerfile

```diff
@@ -0,0 +1,22 @@
+FROM golang:alpine3.17 AS build-env
+
+RUN apk add --update --no-cache git
+
+COPY .  /circuitbreaker 
+
+RUN cd /circuitbreaker \
+    && rm -rf .git \
+    && go install
+RUN chmod a+x $GOPATH/bin/circuitbreaker
```

Comment:
Is this necessary? I thought golang generally produces executables by default.

Reply:
When I tried it for some reason the resulting binary was not executable and the docker command failed.